### ISSUE TITLE
asciinema play fails in xterm

### ISSUE THREAD

**Issue by myitcv** [asciinema/asciinema#367]
I recorded the following using `XTerm(330)`. Playback works in `rxvt-unicode (urxvt) v9.22` but fails in `xterm` itself; things "hang" after Vim loads and no text is inserted.

Any ideas on what's going on would be greatly appreciated.

```
{"version": 2, "width": 319, "height": 83, "timestamp": 1566829577, "env": {"SHELL": "/bin/bash", "TERM": "xterm-256color"}}
[0.400352, "o", "\u001b[38;5;74m- \u001b[38;5;74m\u001b[38;5;101m/home/myitcv \u001b[38;5;74m-\r\r\n\u001b[38;5;74m$ \u001b[0m"]
[1.10629, "o", "v"]
[1.202377, "o", "i"]
[1.410435, "o", "\r\n"]
[1.461325, "o", "\u001b[?1006l\u001b[?1002l"]
[1.477239, "o", "\u001b[?1006h\u001b[?1003h\u001b[?1049h\u001b[22;0;0t\u001b[?1h\u001b=\u001b[?2004h\u001b[1;83r\u001b[?12h\u001b[?12l\u001b[22;2t\u001b[22;1t"]
[1.512668, "o", "\u001b[27m\u001b[23m\u001b[29m\u001b[m\u001b[38;5;251m\u001b[H\u001b[2J"]
[1.524398, "o", "\u001b[2;1H▽\u001b[6n\u001b[2;1H  \u001b[1;1H\u001b[>c\u001b]10;?\u0007\u001b]11;?\u0007"]
[1.525615, "o", "\u001b[?25l\u001b[1;1H\u001b[38;5;242m  1 \r\n~                                                                                                                                                                                                                                                                                                                              \u001b[3;1H~                                                                                                                                                                                                                                                                                                                              \u001b[4;1H~                                                                                                                                                                                                                                                                                                                              \u001b[5;1H~                   "]
[1.525794, "o", "                                                                                                                                                                                                                                                                                                           \u001b[6;1H~                                                                                                                                                                                                                                                                                                                              \u001b[7;1H~                                                                                                                                                                                                                                                                                                                              \u001b[8;1H~                                                                    "]
[1.525861, "o", "                                                                                                                                                                                                                                                          \u001b[9;1H~                                                                                                                                                                                                                                                                                                                              \u001b[10;1H~                                                                                                                                                                                                                                                                                                                              \u001b[11;1H~                                                                                                                   "]
[1.526029, "o", "                                                                                                                                                                                                           \u001b[12;1H~                                                                                                                                                                                                                                                                                                                              \u001b[13;1H~                                                                                                                                                                                                                                                                                                                              \u001b[14;1H~                                                                                                                                                                 "]
[1.526194, "o", "                                                                                                                                                             \u001b[15;1H~                                                                                                                                                                                                                                                                                                                              \u001b[16;1H~                                                                                                                                                                                                                                                                                                                              \u001b[17;1H~                                                                                                                                                                                                               "]
[1.526241, "o", "                                                                                                               \u001b[18;1H~                                                                                                                                                                                                                                                                                                                              \u001b[19;1H~                                                                                                                                                                                                                                                                                                                              \u001b[20;1H~                                                                                                                                                                                                                                                             "]
[1.526381, "o", "                                                                 \u001b[21;1H~                                                                                                                                                                                                                                                                                                                              \u001b[22;1H~                                                                                                                                                                                                                                                                                                                              \u001b[23;1H~                                                                                                                                                                                                                                                                                                           "]
[1.526418, "o", "                   \u001b[24;1H~                                                                                                                                                                                                                                                                                                                              \u001b[25;1H~                                                                                                                                                                                                                                                                                                                              \u001b[26;1H~                                                                                                                                                                                                                                                                                                                              \u001b[27;1H~                   "]
[1.526439, "o", "                                                                                                                                                                                                                                                                                                           \u001b[28;1H~                                                                                                                                                                                                                                                                                                                              \u001b[29;1H~                                                                                                                                                                                                                                                                                                                              \u001b[30;1H~                                                                 "]
[1.526579, "o", "                                                                                                                                                                                                                                                             \u001b[31;1H~                                                                                                                                                                                                                                                                                                                              \u001b[32;1H~                                                                                                                                                                                                                                                                                                                              \u001b[33;1H~                                                                                                               "]
[1.526775, "o", "                                                                                                                                                                                                               \u001b[34;1H~                                                                                                                                                                                                                                                                                                                              \u001b[35;1H~                                                                                                                                                                                                                                                                                                                              \u001b[36;1H~                                                                                                                                                             "]
[1.526973, "o", "                                                                                                                                                                 \u001b[37;1H~                                                                                                                                                                                                                                                                                                                              \u001b[38;1H~                                                                                                                                                                                                                                                                                                                              \u001b[39;1H~                                                                                                                                                                                                           "]
[1.527131, "o", "                                                                                                                   \u001b[40;1H~                                                                                                                                                                                                                                                                                                                              \u001b[41;1H~                                                                                                                                                                                                                                                                                                                              \u001b[42;1H~                                                                                                                                                                                                                                                         "]
[1.527323, "o", "                                                                     \u001b[43;1H~                                                                                                                                                                                                                                                                                                                              \u001b[44;1H~                                                                                                                                                                                                                                                                                                                              \u001b[45;1H~                                                                                                                                                                                                                                                                                                       "]
[1.527503, "o", "                       \u001b[46;1H~                                                                                                                                                                                                                                                                                                                              \u001b[47;1H~                                                                                                                                                                                                                                                                                                                              \u001b[48;1H~                                                                                                                                                                                                                                                                                                                              \u001b[49;1H~               "]
[1.5277, "o", "                                                                                                                                                                                                                                                                                                               \u001b[50;1H~                                                                                                                                                                                                                                                                                                                              \u001b[51;1H~                                                                                                                                                                                                                                                                         "]
[1.527843, "o", "                                                     \u001b[52;1H~                                                                                                                                                                                                                                                                                                                              \u001b[53;1H~                                                                                                                                                                                                                                                                                                                              \u001b[54;1H~                                                                                                                                                                                                                                                                                                                       "]
[1.528016, "o", "       \u001b[55;1H~                                                                                                                                                                                                                                                                                                                              \u001b[56;1H~                                                                                                                                                                                                                                                                                                                              \u001b[57;1H~                                                                                                                                                                                                                                                                                                                              \u001b[58;1H~                               "]
[1.528054, "o", "                                                                                                                                                                                                                                                                                               \u001b[59;1H~                                                                                                                                                                                                                                                                                                                              \u001b[60;1H~                                                                                                                                                                                                                                                                                                                              \u001b[61;1H~                                                                             "]
[1.52819, "o", "                                                                                                                                                                                                                                                 \u001b[62;1H~                                                                                                                                                                                                                                                                                                                              \u001b[63;1H~                                                                                                                                                                                                                                                                                                                              \u001b[64;1H~                                                                                                                           "]
[1.528227, "o", "                                                                                                                                                                                                   \u001b[65;1H~                                                                                                                                                                                                                                                                                                                              \u001b[66;1H~                                                                                                                                                                                                                                                                                                                              \u001b[67;1H~                                                                                                                                                                         "]
[1.528238, "o", "                                                                                                                                                     \u001b[68;1H~                                                                                                                                                                                                                                                                                                                              \u001b[69;1H~                                                                                                                                                                                                                                                                                                                              \u001b[70;1H~                                                                                                                                                                                                                       "]
[1.52835, "o", "                                                                                                       \u001b[71;1H~                                                                                                                                                                                                                                                                                                                              \u001b[72;1H~                                                                                                                                                                                                                                                                                                                              \u001b[73;1H~                                                                                                                                                                                                                                                                     "]
[1.528412, "o", "                                                         \u001b[74;1H~                                                                                                                                                                                                                                                                                                                              \u001b[75;1H~                                                                                                                                                                                                                                                                                                                              \u001b[76;1H~                                                                                                                                                                                                                                                                         "]
[1.529074, "o", "                                                     \u001b[77;1H~                                                                                                                                                                                                                                                                                                                              \u001b[78;1H~                                                                                                                                                                                                                                                                                                                              \u001b[79;1H~                                                                                                                                                                                                                                                                                                                       "]
[1.529157, "o", "       \u001b[80;1H~                                                                                                                                                                                                                                                                                                                              \u001b[81;1H~                                                                                                                                                                                                                                                                                                                              \u001b[m\u001b[38;5;251m\u001b[82;1H\u001b[38;5;236m\u001b[48;5;150m \u001b[m\u001b[38;5;251m\u001b[1m\u001b[38;5;236m\u001b[48;5;150mNORMAL\u001b[m\u001b[38;5;251m\u001b[38;5;236m\u001b[48;5;150m \u001b[m\u001b[38;5;251m\u001b[38;5;150m\u001b[48;5;238m                                                                                                                                                                                                         "]
[1.535032, "o", "                                                                                       \u001b[m\u001b[38;5;251m\u001b[38;5;236m\u001b[48;5;150m 100% \u001b[m\u001b[38;5;251m\u001b[1m\u001b[38;5;236m\u001b[48;5;150m☰    0/1 ㏑\u001b[m\u001b[38;5;251m\u001b[38;5;236m\u001b[48;5;150m :  1 \u001b[m\u001b[38;5;251m\u001b[36;152HVIM - Vi IMproved\u001b[38;152Hversion 8.1.1918\u001b[39;148Hby Bram Moolenaar et al.\u001b[40;139HVim is open source and freely distributable\u001b[42;148HSponsor Vim development!\u001b[43;137Htype  :help sponsor\u001b[38;5;112m<Enter>\u001b[m\u001b[38;5;251m    for information \u001b[45;137Htype  :q\u001b[38;5;112m<Enter>\u001b[m\u001b[38;5;251m               to exit         \u001b[46;137Htype  :help\u001b[38;5;112m<Enter>\u001b[m\u001b[38;5;251m  or  \u001b[38;5;112m<F1>\u001b[m\u001b[38;5;251m  for on-line help\u001b[47;137Htype  :help version8\u001b[38;5;112m<Enter>\u001b[m\u001b[38;5;251m   for version info\u001b]1;[No Name]\u0007"]
[1.535778, "o", "\u001b[1;5H\u001b[?25h"]
[1.535974, "o", "\u001bP+q436f\u001b\\\u001bP+q6b75\u001b\\\u001bP+q6b64\u001b\\\u001bP+q6b72\u001b\\\u001bP+q6b6c\u001b\\\u001bP+q2332\u001b\\\u001bP+q2334\u001b\\\u001bP+q2569\u001b\\\u001bP+q2a37\u001b\\\u001bP+q6b31\u001b\\\u001bP$q q\u001b\\\u001b[?12$p"]
[1.607524, "o", "\u001b[?25l\u001b[36;152H\u001b[38;5;242m                 \u001b[38;152H                \u001b[39;148H                        \u001b[40;139H                                           \u001b[42;148H                        \u001b[43;137H                                              \u001b[45;137H                                              \u001b[46;137H                                              \u001b[47;137H                                              \u001b[1;5H\u001b[?25h"]
[1.885489, "o", "\u001b[?25l\u001b[m\u001b[38;5;251m\u001b[83;1H\u001b[38;5;178m-- INSERT --\u001b[m\u001b[38;5;251m\u001b[82;1H\u001b[38;5;238m\u001b[48;5;110m \u001b[m\u001b[38;5;251m\u001b[1m\u001b[38;5;238m\u001b[48;5;110mINSERT\u001b[m\u001b[38;5;251m\u001b[38;5;236m\u001b[48;5;150m \u001b[m\u001b[38;5;251m\b\u001b[38;5;238m\u001b[48;5;110m \u001b[m\u001b[38;5;251m\u001b[38;5;110m\u001b[48;5;238m                                                                                                                                                                                                                                                                                                \u001b[m\u001b[38;5;251m\u001b[38;5;238m\u001b[48;5;110m 100% \u001b[m\u001b[38;5;251m\u001b[1m\u001b[38;5;238m\u001b[48;5;110m☰    0\u001b[m\u001b[38;5;251m\u001b[1m\u001b[38;5;236m\u001b[48;5;150m/\u001b[m\u001b[38;5;251m\b\u001b[1m\u001b[38;5;238m\u001b[48;5;110m/1 ㏑\u001b[m\u001b[38;5;251m\u001b[38;5;236m\u001b[48;5;150m \u001b[m\u001b[38;5;251m\b\u001b[38;5;238m\u001b[48;5;110m :  1 \u001b[1;5H\u001b[?25h"]
[2.095345, "o", "\u001b[?25l\u001b[m\u001b[38;5;251mT\u001b[82;9H\u001b[38;5;179m\u001b[48;5;238m [+]                                                                                                                                                                                                                                                                                            \u001b[m\u001b[38;5;251m\u001b[11C\u001b[1m\u001b[38;5;238m\u001b[48;5;110m1/\u001b[m\u001b[38;5;251m\u001b[8C\u001b[38;5;238m\u001b[48;5;110m2\u001b[1;6H\u001b[?25h"]
[2.17115, "o", "\u001b[?25l\u001b[m\u001b[38;5;251mh\u001b[82;318H\u001b[38;5;238m\u001b[48;5;110m3\u001b[1;7H\u001b[?25h"]
[2.236112, "o", "\u001b[?25l\u001b[m\u001b[38;5;251mi\u001b[82;318H\u001b[38;5;238m\u001b[48;5;110m4\u001b[1;8H\u001b[?25h"]
[2.356129, "o", "\u001b[?25l\u001b[m\u001b[38;5;251ms\u001b[82;318H\u001b[38;5;238m\u001b[48;5;110m5\u001b[1;9H\u001b[?25h"]
[2.474444, "o", "\u001b[?2004h"]
[2.476487, "o", "\u001b[?25l\u001b[82;318H6\u001b[1;10H\u001b[?25h"]
[2.56437, "o", "\u001b[?25l\u001b[m\u001b[38;5;251mi\u001b[82;318H\u001b[38;5;238m\u001b[48;5;110m7\u001b[1;11H\u001b[?25h"]
[2.756333, "o", "\u001b[?25l\u001b[m\u001b[38;5;251ms\u001b[82;318H\u001b[38;5;238m\u001b[48;5;110m8\u001b[1;12H\u001b[?25h"]
[2.994571, "o", "\u001b[?2004h"]
[2.996602, "o", "\u001b[?25l\u001b[82;318H9\u001b[1;13H\u001b[?25h"]
[3.148, "o", "\u001b[?25l\u001b[m\u001b[38;5;251ma\u001b[82;317H\u001b[38;5;238m\u001b[48;5;110m10\u001b[1;14H\u001b[?25h"]
[3.250547, "o", "\u001b[?2004h"]
[3.252583, "o", "\u001b[?25l\u001b[82;318H1\u001b[1;15H\u001b[?25h"]
[3.356212, "o", "\u001b[?25l\u001b[m\u001b[38;5;251mt\u001b[82;318H\u001b[38;5;238m\u001b[48;5;110m2\u001b[1;16H\u001b[?25h"]
[3.412674, "o", "\u001b[?25l\u001b[m\u001b[38;5;251me\u001b[82;318H\u001b[38;5;238m\u001b[48;5;110m3\u001b[1;17H\u001b[?25h"]
[3.516154, "o", "\u001b[?25l\u001b[m\u001b[38;5;251ms\u001b[82;318H\u001b[38;5;238m\u001b[48;5;110m4\u001b[1;18H\u001b[?25h"]
[3.620209, "o", "\u001b[?25l\u001b[m\u001b[38;5;251mt\u001b[82;318H\u001b[38;5;238m\u001b[48;5;110m5\u001b[1;19H\u001b[?25h"]
[3.901734, "o", "\u001b[m\u001b[38;5;251m\u001b[83;1H\u001b[K\u001b[1;18H\u001b[?25l"]
[3.916815, "o", "\u001b[82;1H\u001b[38;5;236m\u001b[48;5;150m \u001b[m\u001b[38;5;251m\u001b[1m\u001b[38;5;236m\u001b[48;5;150mNORMAL\u001b[m\u001b[38;5;251m\u001b[38;5;238m\u001b[48;5;110m \u001b[m\u001b[38;5;251m\b\u001b[38;5;236m\u001b[48;5;150m \u001b[288C 100% \u001b[m\u001b[38;5;251m\u001b[1m\u001b[38;5;236m\u001b[48;5;150m☰    1\u001b[m\u001b[38;5;251m\u001b[1m\u001b[38;5;238m\u001b[48;5;110m/\u001b[m\u001b[38;5;251m\b\u001b[1m\u001b[38;5;236m\u001b[48;5;150m/1 ㏑\u001b[m\u001b[38;5;251m\u001b[38;5;238m\u001b[48;5;110m \u001b[m\u001b[38;5;251m\b\u001b[38;5;236m\u001b[48;5;150m : 14 \u001b[1;18H\u001b[?25h"]
[4.218166, "o", "\u001b[?25l\u001b[83;1H\u001b[m\u001b[38;5;251m:\u001b[?2004h\u001b[?25h"]
[4.434049, "o", "q"]
[4.761982, "o", "!"]
[5.513934, "o", "\r"]
[5.5199, "o", "\u001b[?25l\u001b[?1006l\u001b[?1003l\u001b[?1002l\u001b[?2004l\u001b]1;unsharemounts\u0007\u001b[23;2t\u001b[23;1t\u001b[22;2t\u001b[22;1t\u001b[23;2t\u001b[23;1t\u001b[83;1H\u001b[K\u001b[83;1H\u001b[?2004l\u001b[?1l\u001b>\u001b[?25h\u001b[?1049l\u001b[23;0;0t"]
[5.532015, "o", "\u001b[38;5;74m- \u001b[38;5;74m\u001b[38;5;101m/home/myitcv \u001b[38;5;74m-\r\r\n\u001b[38;5;74m$ \u001b[0m"]
[6.753574, "o", "exit\r\n"]
```

**Comment by myitcv** [comment]
I totally failed to properly preface the issue description with a big thank you for what this project does already. Only recently did I stumble across asciinema; it's _highly_ impressive.

**Comment by sickill** [comment]
I tried the above one in few terminals on macOS and it worked fine (there was small visual glitch as my terminal is not as wide as the one you had when recording).

It's very weird that it behaves like that when replayed on the exact same terminal it was recorder on 🤔

**Comment by friedrichromstedt** [comment]
Same problem here:

* Arch Linux
* `XTerm(363)`
* `asciinema-2.0.2`

The problem has existed for about 5 months. I stumbled upon it while trying to produce a supplement for a `vim` Issue. Unfortunately I do not have information which versions of `xterm` and `asciinema` I used that time.

The problem is reproducible on my system:
```
$ asciinema rec <filename>.cast
[asciinema] $ vim
:q
[asciinema] $ ^D
$ asciinema play <filename>.cast
```
The playback process hangs after displaying the vim welcome screen and needs to be killed by `$ kill <PID>`. It does not respond to `^C`. After that, the xterm often needs to be `$ reset` when the cursor is left invisible (at least this is the behaviour I found while reproducing the problem right now).

For my use case, I'd like to check the asciinema cast locally using `$ asciinema play <filename>.cast` prior to uploading it. So I'd appreciate any response on how I can help to track this issue down or at least how to work around it.

**Comment by sickill** [comment]
This has been fixed in 2.1.0.

**Comment by friedrichromstedt** [comment]
Confirmed with `xterm(372)` and `asciinema-2.1.0`.

I've tried to verify that the fix is truly due to the `asciinema` upgrade, so I downgraded to `asciinema-2.0.2`, but then running `$ asciinema rec v2.0.2.rec` failed with:
```
Traceback (most recent call last):
  File "/usr/bin/asciinema", line 33, in <module>
    sys.exit(load_entry_point('asciinema==2.0.2', 'console_scripts', 'asciinema')())
  File "/usr/bin/asciinema", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 957, in distribution
    return Distribution.from_name(distribution_name)
  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 548, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for asciinema
```
Just for the records. This is on Arch linux.
